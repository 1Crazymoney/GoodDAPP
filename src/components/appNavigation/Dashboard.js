import React from 'react'
import { View, Text } from 'react-native'
import TabsView from './TabsView'

export default props => {
  const { screenProps } = props
  return (
    <View>
      <TabsView goTo={props.navigation.navigate} routes={screenProps.routes} />
      <Text>Dashboard</Text>
    </View>
  )
}
