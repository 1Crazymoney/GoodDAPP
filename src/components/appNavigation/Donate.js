import React from 'react'
import { View, Text } from 'react-native'
import { createStackNavigator } from './stackNavigation'

const Donate = () => (
  <View>
    <Text>Donate</Text>
  </View>
)

export default createStackNavigator({
  Donate
})
