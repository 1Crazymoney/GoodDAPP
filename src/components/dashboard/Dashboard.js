// @flow
import React, { useEffect, useState } from 'react'
import { StyleSheet, View } from 'react-native'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import { Portal } from 'react-native-paper'
import type { Store } from 'undux'

import GDStore from '../../lib/undux/GDStore'
import SimpleStore from '../../lib/undux/SimpleStore'
import { useDialog } from '../../lib/undux/utils/dialog'
import { getInitialFeed, getNextFeed, PAGE_SIZE } from '../../lib/undux/utils/feed'
import { executeWithdraw } from '../../lib/undux/utils/withdraw'
import { weiToMask } from '../../lib/wallet/utils'
import { createStackNavigator } from '../appNavigation/stackNavigation'
import { PushButton } from '../appNavigation/PushButton'
import TabsView from '../appNavigation/TabsView'
import { Avatar, BigGoodDollar, ClaimButton, Section, Wrapper } from '../common'
import logger from '../../lib/logger/pino-logger'
import userStorage from '../../lib/gundb/UserStorage'
import { PrivacyPolicy, Support, TermsOfUse } from '../webView/webViewInstances'
import Amount from './Amount'
import Claim from './Claim'
import FaceRecognition from './FaceRecognition/FaceRecognition'
import FeedList from './FeedList'
import FeedModalItem from './FeedItems/FeedModalItem'
import Reason from './Reason'
import Receive from './Receive'
import ReceiveFrom from './ReceiveFrom'
import ReceiveAmount from './ReceiveAmount'
import SendByQR from './SendByQR'
import ReceiveByQR from './ReceiveByQR'
import Send from './Send'
import SendConfirmation from './SendConfirmation'
import SendLinkSummary from './SendLinkSummary'
import SendQRSummary from './SendQRSummary'

const log = logger.child({ from: 'Dashboard' })

export type DashboardProps = {
  screenProps: any,
  navigation: any,
  store: Store,
}

type DashboardState = {
  horizontal: boolean,
  feeds: any[],
  currentFeedProps: any,
}

const Dashboard = props => {
  const store = SimpleStore.useStore()
  const gdstore = GDStore.useStore()
  const [showDialog, hideDialog] = useDialog()
  const [state: DashboardState, setState] = useState({
    horizontal: false,
    currentFeedProps: null,
    feeds: [],
  })
  const { params } = props.navigation.state

  useEffect(() => {
    log.debug('Dashboard didmount')
    userStorage.feed.get('byid').on(data => {
      log.debug('gun getFeed callback', { data })
      getFeeds()
    }, true)
  }, [])

  useEffect(() => {
    log.debug('handle links effect dashboard', { params })
    if (params && params.paymentCode) {
      handleWithdraw()
    } else if (params && params.event) {
      showNewFeedEvent(params.event)
    }
  }, [params])

  const getFeeds = () => {
    getInitialFeed(gdstore)
  }

  const showEventModal = item => {
    setState({
      currentFeedProps: {
        item,
        styles: {
          flex: 1,
          alignSelf: 'flex-start',
          height: '100vh',
          position: 'absolute',
          width: '100%',
          paddingTop: normalize(30),
          paddingBottom: normalize(30),
          paddingLeft: normalize(10),
          paddingRight: normalize(10),
          backgroundColor: 'rgba(0, 0, 0, 0.7)',
        },
        onPress: closeFeedEvent,
      },
    })
  }

  const handleFeedSelection = (receipt, horizontal) => {
    showEventModal(receipt)
  }

  const showNewFeedEvent = async eventId => {
    try {
      const item = await userStorage.getFormatedEventById(eventId)
      log.info('showNewFeedEvent', { eventId, item })
      if (item) {
        showEventModal(item)
      } else {
        showDialog({
          title: 'Error',
          message: 'Event does not exist',
        })
      }
    } catch (e) {
      showDialog({
        title: 'Error',
        message: 'Event does not exist',
      })
    }
  }

  const closeFeedEvent = () => {
    setState({ currentFeedProps: null })
  }

  const handleWithdraw = async () => {
    const { paymentCode, reason } = props.navigation.state.params
    try {
      showDialog({ title: 'Processing Payment Link...', loading: true, dismissText: 'hold' })
      await executeWithdraw(store, paymentCode, reason)
      hideDialog()

      // if (receipt.transactionHash) {
      //   await showNewFeedEvent(receipt.transactionHash)
      // }
    } catch (e) {
      showDialog({ title: 'Error', message: e.message })
    }
  }

  const { horizontal, currentFeedProps } = state
  const { screenProps, navigation }: DashboardProps = props
  const { balance, entitlement } = gdstore.get('account')
  const { avatar, fullName } = gdstore.get('profile')
  const feeds = gdstore.get('feeds')

  // TODO: Calculate scroll position to update Dashboard avatar, name and gd amount view
  const scrollPos = 100

  log.info('LOGGER FEEDS', { feeds })
  return (
    <View style={styles.dashboardView}>
      <TabsView goTo={navigation.navigate} routes={screenProps.routes} />
      <Wrapper backgroundColor="#EEE">
        <Section>
          {scrollPos < 100 ? (
            <>
              <Section.Row justifyContent="center" alignItems="baseline">
                <Avatar size={80} source={avatar} onPress={() => screenProps.push('Profile')} />
              </Section.Row>
              <Section.Row justifyContent="center" alignItems="baseline">
                <Section.Title>{fullName || ' '}</Section.Title>
              </Section.Row>
              <Section.Row justifyContent="center" alignItems="baseline">
                <BigGoodDollar number={balance} />
              </Section.Row>
            </>
          ) : (
            <Section.Row>
              <Section.Stack alignItems="flex-start">
                <Avatar size={42} source={avatar} onPress={() => screenProps.push('Profile')} />
              </Section.Stack>
              <Section.Stack alignItems="flex-end">
                <BigGoodDollar number={balance} />
              </Section.Stack>
            </Section.Row>
          )}
          <Section.Row style={styles.buttonsRow} alignItems="stretch">
            <PushButton
              routeName={'Send'}
              screenProps={screenProps}
              style={styles.leftButton}
              icon="send"
              iconAlignment="left"
            >
              Send
            </PushButton>
            <ClaimButton screenProps={screenProps} amount={weiToMask(entitlement, { showUnits: true })} />
            <PushButton
              routeName={'Receive'}
              screenProps={screenProps}
              style={styles.rightButton}
              icon="receive"
              iconAlignment="right"
            >
              Receive
            </PushButton>
          </Section.Row>
        </Section>
        <View style={styles.marginTop}>
          <FeedList
            horizontal={horizontal}
            handleFeedSelection={handleFeedSelection}
            fixedHeight
            virtualized
            data={feeds}
            updateData={() => {}}
            initialNumToRender={PAGE_SIZE}
            onEndReached={getNextFeed.bind(null, store)}
          />
        </View>
        {currentFeedProps && (
          <Portal>
            <FeedModalItem {...currentFeedProps} />
          </Portal>
        )}
      </Wrapper>
    </View>
  )
}

const styles = StyleSheet.create({
  buttonsRow: {
    marginVertical: normalize(8),
  },
  leftButton: {
    flex: 1,
    marginRight: normalize(16),
    paddingRight: normalize(16),
  },
  rightButton: {
    flex: 1,
    marginLeft: normalize(16),
    paddingLeft: normalize(16),
  },
  dashboardView: {
    flex: 1,
  },
  marginTop: {
    marginTop: normalize(8),
  },
  centering: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 8,
    height: '256px',
  },
})

Dashboard.navigationOptions = {
  navigationBarHidden: true,
  title: 'Home',
}

export default createStackNavigator({
  Home: Dashboard,
  Claim,
  Receive,
  ReceiveFrom,
  Amount,
  Reason,
  ReceiveAmount,
  Send,
  SendLinkSummary,
  SendConfirmation,
  FaceRecognition,
  SendByQR,
  ReceiveByQR,
  SendQRSummary,
  PP: PrivacyPolicy,
  TOU: TermsOfUse,
  Support,
})
