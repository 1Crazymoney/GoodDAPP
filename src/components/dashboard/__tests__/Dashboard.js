import React from 'react'
import Dashboard from '../Dashboard'
import Donate from '../../appNavigation/Donate'
import { createSwitchNavigator } from '@react-navigation/core'
import { createBrowserApp } from '@react-navigation/web'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe.skip('Dashboard', () => {
  it('renders without errors', () => {
    const routes = {
      Dashboard,
      Donate
    }

    const AppNavigator = createBrowserApp(createSwitchNavigator(routes))
    class AppNavigation extends React.Component<AppNavigationProps, AppNavigationState> {
      static router = AppNavigator.router

      render() {
        return <AppNavigator navigation={this.props.navigation} screenProps={{ routes }} />
      }
    }

    const tree = renderer.create(<AppNavigation />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const WebRouter = createBrowserApp(createSwitchNavigator({ Dashboard }))

    const component = renderer.create(<WebRouter />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot when changing tab', () => {
    const WebRouter = createBrowserApp(createSwitchNavigator({ Dashboard }))
    const component = renderer.create(<WebRouter />)
    const [tabsView] = component.toJSON()
    const [tabButton] = tabsView.children
    tabButton.props.onClick(new Event('fakeEvent'))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
