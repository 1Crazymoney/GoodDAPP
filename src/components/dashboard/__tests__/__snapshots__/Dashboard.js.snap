// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n r-flex-13awgt0 r-pointerEvents-12vffkv"
>
  <div
    className=""
    style={
      Object {
        "bottom": 0,
        "display": "flex",
        "flexDirection": "column",
        "flexGrow": 1,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <div
      className="css-view-1dbjc4n r-paddingBottom-97e31f r-paddingLeft-bv2aro r-paddingRight-hxflta r-paddingTop-10xqauy"
      style={
        Object {
          "backgroundColor": "rgba(0,175,255,1.00)",
          "boxShadow": "0px 3px 4px rgba(0,0,0,0.24)",
          "zIndex": 0,
        }
      }
    >
      <div
        className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz r-overflow-1udh08x"
        style={
          Object {
            "backgroundColor": "rgba(0,175,255,1.00)",
            "height": "56px",
            "paddingLeft": "4px",
            "paddingRight": "4px",
          }
        }
      >
        <div
          className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "WebkitAlignItems": "flex-start",
              "WebkitBoxAlign": "start",
              "alignItems": "flex-start",
              "marginLeft": "10px",
              "msFlexAlign": "start",
              "width": "37px",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao"
            data-testid="invite_tab"
            dir="auto"
            style={
              Object {
                "color": "rgba(255,255,255,1.00)",
                "fontFamily": "gooddollar",
                "fontSize": "36px",
                "fontStyle": "normal",
                "fontWeight": "normal",
              }
            }
          >
            
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-cursor-1loqt21 r-flex-13awgt0 r-paddingHorizontal-1e081e0 r-touchAction-1otgn73"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          tabIndex="0"
        >
          <h1
            className="css-reset-4rbku5 css-text-901oao css-textOneLine-bfa6kz r-fontSize-adyw6z"
            dir="auto"
            role="heading"
            style={
              Object {
                "color": "rgba(255,255,255,1.00)",
                "fontFamily": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
                "fontWeight": "500",
                "textAlign": "left",
              }
            }
          />
        </div>
        <div
          className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          data-focusable={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "WebkitAlignItems": "flex-end",
              "WebkitBoxAlign": "end",
              "alignItems": "flex-end",
              "msFlexAlign": "end",
              "width": "37px",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-text-901oao r-fontSize-adyw6z"
            data-testid="burger_button"
            dir="auto"
            style={
              Object {
                "color": "rgba(255,255,255,1.00)",
                "fontFamily": "gooddollar",
                "fontStyle": "normal",
                "fontWeight": "normal",
                "marginRight": "10px",
              }
            }
          >
            
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n r-position-bnwqim"
      data-name="viewWrapper"
      style={
        Object {
          "WebkitBoxDirection": "normal",
          "WebkitBoxFlex": 1,
          "WebkitBoxOrient": "vertical",
          "WebkitFlexDirection": "column",
          "WebkitFlexGrow": 1,
          "backgroundColor": "rgba(238,238,238,1.00)",
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": 1,
          "maxHeight": "844px",
          "msFlexDirection": "column",
          "msFlexPositive": 1,
          "paddingBottom": "0px",
          "paddingLeft": "0px",
          "paddingRight": "0px",
          "paddingTop": "0px",
          "width": "100%",
        }
      }
    >
      <div
        className="css-view-1dbjc4n r-paddingHorizontal-1e081e0 r-position-bnwqim"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1.00)",
            "borderBottomLeftRadius": "5px",
            "borderBottomRightRadius": "5px",
            "borderTopLeftRadius": "0px",
            "borderTopRightRadius": "0px",
            "marginBottom": "-3px",
            "marginLeft": "8px",
            "marginRight": "8px",
            "paddingBottom": "8px",
            "paddingLeft": "8px",
            "paddingRight": "8px",
            "paddingTop": "16px",
            "zIndex": 10,
          }
        }
      >
        <div
          className="css-view-1dbjc4n r-position-bnwqim"
          style={
            Object {
              "height": "165px",
            }
          }
        >
          <div
            className="css-view-1dbjc4n r-alignItems-1awozwy"
            style={
              Object {
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "vertical",
                "WebkitBoxPack": "justify",
                "WebkitFlexDirection": "column",
                "WebkitJustifyContent": "space-between",
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "space-between",
                "msFlexDirection": "column",
                "msFlexPack": "justify",
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "height": "68px",
                  "left": "187.5px",
                  "position": "absolute",
                  "top": "0px",
                  "width": "68px",
                }
              }
            >
              <div
                className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionDuration-eafdt9 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
                data-focusable={true}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
                tabIndex="0"
              >
                <div
                  className="css-view-1dbjc4n"
                  style={
                    Object {
                      "borderBottomLeftRadius": "50%",
                      "borderBottomRightRadius": "50%",
                      "borderTopLeftRadius": "50%",
                      "borderTopRightRadius": "50%",
                      "height": "100%",
                      "width": "100%",
                    }
                  }
                >
                  <unknownProfile.svg />
                </div>
              </div>
            </div>
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-justifyContent-1777fci"
              style={
                Object {
                  "bottom": "0px",
                  "marginBottom": "auto",
                  "marginTop": "auto",
                  "opacity": 1,
                  "paddingTop": "10px",
                  "position": "absolute",
                  "top": "0px",
                  "zIndex": -1,
                }
              }
            >
              <div
                className="css-text-901oao"
                dir="auto"
                style={
                  Object {
                    "color": "rgba(163,163,163,1.00)",
                    "fontFamily": "Roboto Slab",
                    "fontSize": "18px",
                    "fontWeight": "normal",
                    "lineHeight": "24px",
                    "textAlign": "center",
                    "textDecoration": "none",
                    "textTransform": "none",
                  }
                }
              >
                 
              </div>
            </div>
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitAlignItems": "baseline",
                  "WebkitBoxAlign": "baseline",
                  "alignItems": "baseline",
                  "bottom": "0px",
                  "marginBottom": "16px",
                  "marginTop": "16px",
                  "msFlexAlign": "baseline",
                  "position": "absolute",
                  "right": "187.5px",
                  "visibility": "hidden",
                }
              }
            >
              <div
                className="css-view-1dbjc4n"
              >
                <div
                  className="css-view-1dbjc4n r-flexDirection-18u37iz"
                  data-testid="amount_value"
                  style={
                    Object {
                      "WebkitAlignItems": "baseline",
                      "WebkitBoxAlign": "baseline",
                      "alignItems": "baseline",
                      "display": "flex",
                      "msFlexAlign": "baseline",
                    }
                  }
                >
                  <div
                    className="css-text-901oao"
                    dir="auto"
                    style={
                      Object {
                        "color": "rgba(85,85,85,1.00)",
                        "fontFamily": "Roboto Slab",
                        "fontSize": "42px",
                        "fontWeight": "600",
                        "lineHeight": "42px",
                        "marginRight": "2px",
                        "textAlign": "left",
                        "textDecoration": "none",
                        "textTransform": "none",
                      }
                    }
                  >
                    -.--
                  </div>
                  <div
                    className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
                    style={
                      Object {
                        "WebkitBoxPack": "justify",
                        "WebkitJustifyContent": "space-between",
                        "justifyContent": "space-between",
                        "marginRight": "-20px",
                        "msFlexPack": "justify",
                      }
                    }
                  >
                    <div
                      className="css-text-901oao"
                      dir="auto"
                      style={
                        Object {
                          "color": "rgba(85,85,85,1.00)",
                          "fontFamily": "Roboto Slab",
                          "fontSize": "18px",
                          "fontWeight": "700",
                          "lineHeight": "22px",
                          "textAlign": "right",
                          "textDecoration": "none",
                          "textTransform": "none",
                        }
                      }
                    >
                      G
                    </div>
                    <div
                      className="css-text-901oao"
                      dir="auto"
                      style={
                        Object {
                          "color": "rgba(85,85,85,1.00)",
                          "fontFamily": "Roboto Slab",
                          "fontSize": "17.5px",
                          "fontWeight": "700",
                          "lineHeight": "22px",
                          "textAlign": "right",
                          "textDecoration": "none",
                          "textTransform": "none",
                        }
                      }
                    >
                      $
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-18u37iz"
          style={
            Object {
              "WebkitBoxPack": "justify",
              "WebkitJustifyContent": "space-between",
              "height": "70px",
              "justifyContent": "space-between",
              "marginBottom": "0px",
              "marginTop": "1px",
              "msFlexPack": "justify",
            }
          }
        >
          <div
            className="css-view-1dbjc4n r-flex-13awgt0 r-justifyContent-1777fci"
            style={
              Object {
                "WebkitAlignItems": "stretch",
                "WebkitBoxAlign": "stretch",
                "alignItems": "stretch",
                "backgroundColor": "rgba(0,175,255,1.00)",
                "borderBottomColor": "rgba(0,175,255,1.00)",
                "borderBottomLeftRadius": "50px",
                "borderBottomRightRadius": "50px",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0px",
                "borderLeftColor": "rgba(0,175,255,1.00)",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0px",
                "borderRightColor": "rgba(0,175,255,1.00)",
                "borderRightStyle": "solid",
                "borderRightWidth": "0px",
                "borderTopColor": "rgba(0,175,255,1.00)",
                "borderTopLeftRadius": "50px",
                "borderTopRightRadius": "50px",
                "borderTopStyle": "solid",
                "borderTopWidth": "0px",
                "display": "flex",
                "height": "44px",
                "marginBottom": "0px",
                "marginLeft": "0px",
                "marginRight": "-12px",
                "marginTop": "0px",
                "minHeight": "44px",
                "minWidth": "auto",
                "msFlexAlign": "stretch",
                "paddingBottom": "0px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "paddingTop": "0px",
              }
            }
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-cursor-1loqt21 r-flex-13awgt0 r-overflow-1udh08x r-position-bnwqim r-touchAction-1otgn73"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              style={
                Object {
                  "WebkitAlignSelf": "stretch",
                  "alignSelf": "stretch",
                  "borderBottomLeftRadius": "50px",
                  "borderBottomRightRadius": "50px",
                  "borderTopLeftRadius": "50px",
                  "borderTopRightRadius": "50px",
                  "msFlexItemAlign": "stretch",
                }
              }
              tabIndex="0"
            >
              <div
                className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci"
                style={
                  Object {
                    "WebkitAlignItems": "stretch",
                    "WebkitAlignSelf": "stretch",
                    "WebkitBoxAlign": "stretch",
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "msFlexAlign": "stretch",
                    "msFlexItemAlign": "stretch",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-marginHorizontal-1ybube5 r-marginVertical-712mwr"
                >
                  <div
                    className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-flexDirection-18u37iz"
                    style={
                      Object {
                        "display": "flex",
                      }
                    }
                  >
                    <div
                      className="css-text-901oao r-fontSize-adyw6z"
                      dir="auto"
                      style={
                        Object {
                          "color": "rgba(255,255,255,1.00)",
                          "fontFamily": "gooddollar",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                          "marginRight": "16px",
                        }
                      }
                    >
                      
                    </div>
                    <div
                      className="css-text-901oao r-alignItems-1awozwy r-justifyContent-1777fci"
                      dir="auto"
                      style={
                        Object {
                          "color": "rgba(255,255,255,1.00)",
                          "fontFamily": "Roboto",
                          "fontSize": "16px",
                          "fontWeight": "500",
                          "letterSpacing": "0px",
                          "lineHeight": "22px",
                          "marginRight": "16px",
                          "paddingTop": "1px",
                          "textAlign": "center",
                          "textDecoration": "none",
                          "textTransform": "uppercase",
                        }
                      }
                    >
                      Send
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "zIndex": 1,
              }
            }
          >
            <div
              className="css-view-1dbjc4n"
              style={
                Object {
                  "WebkitTransform": "scale(1)",
                  "height": "72px",
                  "transform": "scale(1)",
                  "width": "72px",
                }
              }
            >
              <div
                className="css-view-1dbjc4n r-alignItems-1awozwy r-justifyContent-1777fci"
                data-testid="claim_button"
                style={
                  Object {
                    "backgroundColor": "rgba(0,195,174,1.00)",
                    "borderBottomColor": "rgba(255,255,255,1.00)",
                    "borderBottomLeftRadius": "50%",
                    "borderBottomRightRadius": "50%",
                    "borderBottomStyle": "solid",
                    "borderBottomWidth": "3px",
                    "borderLeftColor": "rgba(255,255,255,1.00)",
                    "borderLeftStyle": "solid",
                    "borderLeftWidth": "3px",
                    "borderRightColor": "rgba(255,255,255,1.00)",
                    "borderRightStyle": "solid",
                    "borderRightWidth": "3px",
                    "borderTopColor": "rgba(255,255,255,1.00)",
                    "borderTopLeftRadius": "50%",
                    "borderTopRightRadius": "50%",
                    "borderTopStyle": "solid",
                    "borderTopWidth": "3px",
                    "display": "flex",
                    "height": "100%",
                    "marginBottom": "0px",
                    "marginLeft": "0px",
                    "marginRight": "0px",
                    "marginTop": "0px",
                    "minHeight": "44px",
                    "minWidth": "64px",
                    "paddingBottom": "3px",
                    "paddingLeft": "0px",
                    "paddingRight": "0px",
                    "paddingTop": "3px",
                    "position": "absolute",
                    "width": "100%",
                    "zIndex": 99,
                  }
                }
              >
                <div
                  className="css-cursor-18t94o4 css-view-1dbjc4n r-cursor-1loqt21 r-flex-13awgt0 r-overflow-1udh08x r-position-bnwqim r-touchAction-1otgn73"
                  data-focusable={true}
                  disabled={false}
                  onKeyDown={[Function]}
                  onKeyPress={[Function]}
                  onKeyUp={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  role="button"
                  style={
                    Object {
                      "WebkitAlignSelf": "stretch",
                      "alignSelf": "stretch",
                      "borderBottomLeftRadius": "50%",
                      "borderBottomRightRadius": "50%",
                      "borderTopLeftRadius": "50%",
                      "borderTopRightRadius": "50%",
                      "msFlexItemAlign": "stretch",
                    }
                  }
                  tabIndex="0"
                >
                  <div
                    className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci"
                    style={
                      Object {
                        "WebkitAlignSelf": "stretch",
                        "alignSelf": "stretch",
                        "marginLeft": "-16px",
                        "marginRight": "-16px",
                        "msFlexItemAlign": "stretch",
                      }
                    }
                  >
                    <div
                      className="css-view-1dbjc4n r-marginHorizontal-1ydw1k6 r-marginVertical-712mwr"
                    >
                      <div
                        className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-flexDirection-18u37iz"
                        style={
                          Object {
                            "display": "flex",
                          }
                        }
                      >
                        <div
                          className="css-text-901oao r-alignItems-1awozwy r-justifyContent-1777fci"
                          dir="auto"
                          style={
                            Object {
                              "color": "rgba(255,255,255,1.00)",
                              "fontFamily": "Roboto",
                              "fontSize": "16px",
                              "fontWeight": "500",
                              "letterSpacing": "0px",
                              "lineHeight": "22px",
                              "paddingTop": "1px",
                              "textAlign": "center",
                              "textDecoration": "none",
                              "textTransform": "uppercase",
                            }
                          }
                        >
                          Claim
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="css-view-1dbjc4n r-flex-13awgt0 r-justifyContent-1777fci"
            style={
              Object {
                "WebkitAlignItems": "stretch",
                "WebkitBoxAlign": "stretch",
                "alignItems": "stretch",
                "backgroundColor": "rgba(0,175,255,1.00)",
                "borderBottomColor": "rgba(0,175,255,1.00)",
                "borderBottomLeftRadius": "50px",
                "borderBottomRightRadius": "50px",
                "borderBottomStyle": "solid",
                "borderBottomWidth": "0px",
                "borderLeftColor": "rgba(0,175,255,1.00)",
                "borderLeftStyle": "solid",
                "borderLeftWidth": "0px",
                "borderRightColor": "rgba(0,175,255,1.00)",
                "borderRightStyle": "solid",
                "borderRightWidth": "0px",
                "borderTopColor": "rgba(0,175,255,1.00)",
                "borderTopLeftRadius": "50px",
                "borderTopRightRadius": "50px",
                "borderTopStyle": "solid",
                "borderTopWidth": "0px",
                "display": "flex",
                "height": "44px",
                "marginBottom": "0px",
                "marginLeft": "-12px",
                "marginRight": "0px",
                "marginTop": "0px",
                "minHeight": "44px",
                "minWidth": "auto",
                "msFlexAlign": "stretch",
                "paddingBottom": "0px",
                "paddingLeft": "0px",
                "paddingRight": "0px",
                "paddingTop": "0px",
              }
            }
          >
            <div
              className="css-cursor-18t94o4 css-view-1dbjc4n r-cursor-1loqt21 r-flex-13awgt0 r-overflow-1udh08x r-position-bnwqim r-touchAction-1otgn73"
              data-focusable={true}
              disabled={false}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              role="button"
              style={
                Object {
                  "WebkitAlignSelf": "stretch",
                  "alignSelf": "stretch",
                  "borderBottomLeftRadius": "50px",
                  "borderBottomRightRadius": "50px",
                  "borderTopLeftRadius": "50px",
                  "borderTopRightRadius": "50px",
                  "msFlexItemAlign": "stretch",
                }
              }
              tabIndex="0"
            >
              <div
                className="css-view-1dbjc4n r-flex-13awgt0 r-flexDirection-18u37iz r-justifyContent-1777fci"
                style={
                  Object {
                    "WebkitAlignItems": "stretch",
                    "WebkitAlignSelf": "stretch",
                    "WebkitBoxAlign": "stretch",
                    "alignItems": "stretch",
                    "alignSelf": "stretch",
                    "msFlexAlign": "stretch",
                    "msFlexItemAlign": "stretch",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-marginHorizontal-1ybube5 r-marginVertical-712mwr"
                >
                  <div
                    className="css-view-1dbjc4n r-alignItems-1awozwy r-flex-13awgt0 r-flexDirection-18u37iz"
                    style={
                      Object {
                        "display": "flex",
                      }
                    }
                  >
                    <div
                      className="css-text-901oao r-alignItems-1awozwy r-justifyContent-1777fci"
                      dir="auto"
                      style={
                        Object {
                          "color": "rgba(255,255,255,1.00)",
                          "fontFamily": "Roboto",
                          "fontSize": "16px",
                          "fontWeight": "500",
                          "letterSpacing": "0px",
                          "lineHeight": "22px",
                          "marginLeft": "16px",
                          "paddingTop": "1px",
                          "textAlign": "center",
                          "textDecoration": "none",
                          "textTransform": "uppercase",
                        }
                      }
                    >
                      Receive
                    </div>
                    <div
                      className="css-text-901oao r-fontSize-adyw6z"
                      dir="auto"
                      style={
                        Object {
                          "color": "rgba(255,255,255,1.00)",
                          "fontFamily": "gooddollar",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                          "marginLeft": "16px",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
